vMenu/CommonFunctions.cs

```diff
@@ -680,6 +664,27 @@ public static async Task TeleportToCoords(Vector3 pos, bool safeModeDisabled = f
                     Game.PlayerPed.IsPositionFrozen = false;
                 }
 
+                // Set focus for that area.
+                SetFocusPosAndVel(pos.X, pos.Y, pos.Z, 0, 0, 0);
+
+                // Start to load the scene.
+                NewLoadSceneStart(pos.X, pos.Y, pos.Z, pos.X, pos.Y, pos.Z, 50f, 0);
+
+                // Reset the timer.
+                tempTimer = GetGameTimer();
+
+                // Wait for the scene to be loaded with a timeout.
+                while (!IsNewLoadSceneLoaded() && Game.GameTime - tempTimer < 10000) // Can take up to 10 seconds
```

Comment:
I'm not sure this while loop is in the best place right now, because the player is already teleported and not yet frozen. The reason I added this loading new scene, was to hopefully make the game load the correct Z coordinate for the given coordinates more quickly.

I think stopping the loading scene and reworking the timer is great, but moving it down here probably doesn't have the desired effect.

Reply:
Alright I see what you're saying, I'll rework it.